Develop

python4eg opened the pull request.

python4eg commented on static/js/pages/UserPage/UserPage.js:
```diff
@@ -0,0 +1,26 @@
+    }
+
+    componentDidMount() {
+        fetch(`http://localhost:3090/api/profile`)
```
Should be configurable

python4eg commented on static/js/pages/UserPage/UserPage.js:
```diff
@@ -0,0 +1,26 @@
+
+    componentDidMount() {
+        fetch(`http://localhost:3090/api/profile`)
+            .then(response => response.json() )
```
error handling for json parding

python4eg commented on static/js/pages/UserPage/Result.js:
```diff
@@ -0,0 +1,63 @@
+
+class ResultItem extends React.Component{
+    render(){
+        var user = this.props.user;
```
PropTypes

python4eg commented on static/js/pages/UserPage/Result.js:
```diff
@@ -0,0 +1,63 @@
+class ResultItem extends React.Component{
+    render(){
+        var user = this.props.user;
+        user.is_premium == false ? user.is_premium = "none" : user.is_premium = "active";
```
why?

python4eg commented on static/js/pages/UserPage/Result.js:
```diff
@@ -0,0 +1,63 @@
+    padding: '20px',
+    textAlign: 'center'
+}
+
```
too many spaces

python4eg commented on static/js/pages/RecipesPage/RecipeItem.js:
```diff
@@ -0,0 +1,64 @@
+import { Link } from 'react-router';
+import { Grid, Row, Col } from 'react-bootstrap';
+
+const imgStyle = {
```
why don't you put it into css?

python4eg commented on static/js/pages/SignInPage/SignInForm.js:
```diff
@@ -0,0 +1,114 @@
+        credentials.email = this.state.email;
+        credentials.password = this.state.password;
+
+        axios.post('/api/login', credentials)
```
https://en.wikipedia.org/wiki/Basic_access_authentication

l1s4un committed:
Move SearchComponent to RecipesPage

Add searchIcon to SearchBar

apasashkov committed:
- sending hashed password to server;
- removed Object.assign for state;
- error styling change;

iuriifaevskii committed:
create an opportunity to search recipe by name (test version)

iuriifaevskii committed:
bugfix in constructor

l1s4un committed:
add DropDown component

l1s4un committed:
add DropDown component

l1s4un committed:
Merge branch 'feature/search_recipe' of https://github.com/Social-projects-Rivne/Rv-026WebUI into feature/search_recipe

l1s4un committed:
add default style to DropDown component

iuriifaevskii committed:
implement dropdown logic

iuriifaevskii committed:
fix conflict

iuriifaevskii committed:
fix bug from correct search

iuriifaevskii committed:
add search by tag type

l1s4un committed:
add component SearchButton

apasashkov committed:
Recipe id validation change.

apasashkov committed:
merge with ShowRecipe

apasashkov committed:
added json-loader

iuriifaevskii committed:
add completely search logic using the button

apasashkov committed:
- signUp and signIn form submit disabled when fields are not validated;
- changed server errors o signIn;
- hashing password on the client;
- removed Object.assign in setState
-

apasashkov committed:
- corrected createrecipe route

iuriifaevskii committed:
change style to search bar

iuriifaevskii committed:
fix search recipe model

StasMelnyk committed:
save changes in db config

save changes in db config

l1s4un committed:
Change SearchElement and SearchElements components styling

iuriifaevskii committed:
add proptypes to search components

iuriifaevskii committed:
Merge branch 'feature/search_recipe' of https://github.com/Social-projects-Rivne/Rv-026WebUI into feature/search_recipe

l1s4un committed:
Change footer position to relative

apasashkov committed:
Merge pull request #22 from Social-projects-Rivne/feature/showRecipe2

Feature/show recipe2

apasashkov committed:
removed unnecessary console.log

l1s4un committed:
add SearchElementRating with stars to SearchElement Component

iuriifaevskii committed:
Merge branch 'feature/search_recipe' of https://github.com/Social-projects-Rivne/Rv-026WebUI into feature/search_recipe

iuriifaevskii committed:
Merge branch 'feature/search_recipe' of https://github.com/Social-projects-Rivne/Rv-026WebUI into feature/search_recipe

apasashkov committed:
Changes in users db table:
- added is_activated, activation_id fields;
- changed is_deleted from boolean to date;
- appropriate changes in dbseed.js

apasashkov committed:
- changed is_deleted from boolean to date;
- changed sign up logic with sending a unique activation_id link to user and saving it to db with is_activated=false field;
- setting is_activated=true in db upon entering the link via email;

ltlmgw committed:
сreated the ability to edit your own profile

ltlmgw committed:
small changes in checkUserId

ltlmgw committed:
changes in checkUserId

ltlmgw committed:
added in app routes

ltlmgw committed:
some images changes

Commit:
deleted config.js

apasashkov committed:
Merge pull request #23 from Social-projects-Rivne/signup_change

Signup change

StasMelnyk committed:
Merge remote-tracking branch 'origin/develop' into feature/editRecipe

StasMelnyk committed:
clean project

clean project

StasMelnyk committed:
merge from develop

merge from develop

apasashkov committed:
added main page

apasashkov committed:
Merge branch 'develop' of https://github.com/Social-projects-Rivne/Rv-026WebUI into develop
merge

apasashkov committed:
- added login/logout Header display depending on loggedIn status;
- moved Header.js from layouts to common;
- added Header tag to all pages;

apasashkov committed:
Merge pull request #25 from Social-projects-Rivne/feature/login_header

- added login/logout Header display depending on loggedIn status;

iuriifaevskii committed:
add new responsive style for recipes

apasashkov committed:
added role_id=1 when registering a new user

iuriifaevskii committed:
rename recipe to recipes,fix style of recipe,add link to one recipe

iuriifaevskii committed:
fix search styles

StasMelnyk committed:
add edit recipe

added possibility create new ingredients\tags
added possibility add or delete connection in calc_card/tag_recipes
changed fields resipe_id to recipe_id
changed is_deleted format from boolean to date
added edit recipe fields(photo, title, description)

iuriifaevskii committed:
change the search logic to the GET method, move the search folder in recipePage folder

iuriifaevskii committed:
merge with dev

iuriifaevskii committed:
refactor code by using eslint

iuriifaevskii committed:
add new style for create recipe form

StasMelnyk committed:
added checking  holder of the recipe

added checking
holder of the recipe
added link for tags

iuriifaevskii committed:
fix database issues

StasMelnyk committed:
cleaned up project from eslint error

cleaned up project from eslint error
rename addConnect/deleteConnect to addDbLink/removeDbLink

StasMelnyk committed:
hot fix

hot fix

ltlmgw committed:
Added hover effect on inline-edit elements. User cant edit email and also take admin role now.

ltlmgw committed:
Merge branch 'feature/profile_edit' of https://github.com/Social-projects-Rivne/Rv-026WebUI into feature/profile_edit

apasashkov committed:
changed role id to 2 on sign up

StasMelnyk committed:
Merge branch 'develop' into feature/editRecipe

StasMelnyk committed:
Merge pull request #28 from Social-projects-Rivne/feature/editRecipe

Feature/edit recipe

iuriifaevskii committed:
create loader pre uploading recipes

iuriifaevskii committed:
merge conflict

ltlmgw committed:
resolved conflicts

iuriifaevskii committed:
fix image path

Commit:
Merge pull request #24 from Social-projects-Rivne/feature/profile_edit

Feature/profile edit

Commit:
Update UserPage.js

oppps, sorry. conflict fixed

iuriifaevskii committed:
fix db seed

iuriifaevskii committed:
merge pull request #29

StasMelnyk committed:
fixed bugs

fixed bugs

StasMelnyk committed:
Merge branch 'develop' into feature/editRecipe

StasMelnyk committed:
Merge pull request #30 from Social-projects-Rivne/feature/editRecipe

fixed bugs

StasMelnyk committed:
changed recipe model

changed recipe model

StasMelnyk committed:
hot fix

hot fix

StasMelnyk committed:
Merge pull request #31 from Social-projects-Rivne/feature/editRecipe

changed recipe model

apasashkov committed:
Added the abilty to paste text to signUp and signIn forms

iuriifaevskii committed:
add stars to the labels and redirect after adding recipe

python4g commented on controllers/recipeController.js:
```diff
@@ -0,0 +1,359 @@
+recipeController.createRecipe = (req, res, next) => {
+    let form = new multiparty.Form();
+
+    form.parse(req, (err, fields, files) => {
```
http://webuniverse.io/cyclomatic-complexity-refactoring-tips/

python4g commented on controllers/recipeController.js:
```diff
@@ -0,0 +1,359 @@
+                    const insertId = result.rows[0].id;
+                    db.query(recipeObject.addDbLink(table, fieldConnect, recipeData.id, insertId), (error) => {
+                        if (error) {
+                            status = false;
```
assign status number here

python4g commented on controllers/recipeController.js:
```diff
@@ -0,0 +1,359 @@
+    } else {
+        db.query(recipeObject.upsertData(recipeData), (err) => {
+            if (err) {
+                res.sendStatus(500);
```
assign to `status` var

python4g commented on controllers/recipeController.js:
```diff
@@ -0,0 +1,359 @@
+            } else {
+                res.sendStatus(200);
+            }
+        });
```
send status outside `if..else` statement

python4g commented on db/config.example.js:
```diff
@@ -0,0 +1,13 @@
+var userName = "userName";
```
lowercase

python4g commented on index.html:
```diff
@@ -0,0 +1,13 @@
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/latest/css/bootstrap.min.css" />
```
`type` is missed

python4g commented on static/js/common/EditableImage.js:
```diff
@@ -0,0 +1,155 @@
+                saving: false,
+                editing: false });
+            wait(3000)
+            .then(() => {
```
👍

python4g commented on static/js/common/EditableList.js:
```diff
@@ -0,0 +1,215 @@
+            saving: false,
+            message: '',
+        };
+        this.edit = this.edit.bind(this);
```
what about es6 way??
```javascript
 this.edit = () => this.edit());
```

python4g commented on static/js/common/EditableList.js:
```diff
@@ -0,0 +1,215 @@
+        onSave: PropTypes.func,
+    }
+
+    constructor(props) {
```
what about es6 way with class properties?
```javascript
edit = () => this.edit();
saveEdit = () => this.saveEdit();
cancelEdit = () =>  this.cancelEdit();
constructor(props) {
        super(props);
        this.state = {
            cancelItem: [],
            deletedItem: [],
            items: this.props.data,
            editing: false,
            saving: false,
            message: '',
        };
}
```

python4g commented on static/js/common/EditableList.js:
```diff
@@ -0,0 +1,215 @@
+                );
+            }
+            return ( 
+                <div>
```
nitpick: Inheritance indent for readability 
```javascript
<div>
    <div>
       <ul className = "list-group" > 
           { items } 
       </ul>
   </div>
</div>
```

python4g commented on static/js/common/EditableList.js:
```diff
@@ -0,0 +1,215 @@
+            createItem(e) {
+                e.preventDefault();
+                var item = this.refs.itemName.value;
+                if (typeof item === 'string' && item.length > 0) {
```
https://lodash.com/docs#isString and https://lodash.com/docs/4.17.4#isEmpty

python4g commented on static/js/common/EditableText.js:
```diff
@@ -0,0 +1,182 @@
+
+    renderSavingAnimation() {
+        if (this.state.saving) {
+            return <ReactLoading style={centerDiv} type="bars" color="#444" height="70" width="20" />;
```
move repeatable values into constants.
I've seen `bars` and color `#444` many times

StasMelnyk replied:
This is temporary for now. In the future, we will create new preloader

python4g commented on static/js/common/Header.js:
```diff
@@ -0,0 +1,85 @@
+
+    componentWillMount() {
+        if (document.cookie) {
+            this.setState({ loggedIn: true });
```
`this.setState({ loggedIn: document.cookie});`

python4g commented on static/js/layouts/MainLayout.js:
```diff
@@ -0,0 +1,16 @@
+    render() {
+        return (
+            <div>
+                    {this.props.children}
```
indent

python4g commented on static/js/common/EditableList.js:
```diff
@@ -0,0 +1,215 @@
+
+
+    render() {
+        const items = this.state.items.map(function(item, index) {
```
arrow function

python4g commented on static/js/pages/CreateRecipePage/CreateRecipePage.js:
```diff
@@ -0,0 +1,63 @@
+    addRecipe(recipe) {
+        return new Promise((resolve, reject) => {
+            const data = new FormData();
+            data.append('title', recipe.title);
```
why not just:
```
for ( var key in recipe) {
    data.append(key, recipe[key]);
}
```

python4g commented on static/js/pages/CreateRecipePage/CreateRecipePage.js:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    addRecipe(recipe) {
+        return new Promise((resolve, reject) => {
```
👍  for `Promise` and what about using `axios`?

iuriifaevskii committed:
create component for adding many elements

l1s4un committed:
include SASS scss styling

iuriifaevskii committed:
add style to ListCreator component and refactor some code

iuriifaevskii committed:
add axios instead of the promise

iuriifaevskii committed:
divided createRecipe into several small functions

iuriifaevskii committed:
create functional for adding ingredients

l1s4un committed:
translate scss with gulp

l1s4un committed:
delete vmcode config

iuriifaevskii committed:
fix style for the ListCreator

iuriifaevskii committed:
fix the bug before adding many elements in database

iuriifaevskii committed:
fix styles and add links to header

iuriifaevskii committed:
replaced the class with the function in the recipe model

l1s4un committed:
format config.example.js

l1s4un committed:
add watchers

l1s4un committed:
Merge branch 'feature/migrate_to_Sass_styling' of https://github.com/Social-projects-Rivne/Rv-026WebUI into feature/migrate_to_Sass_styling
some changes)

l1s4un committed:
fix errors

l1s4un committed:
Merge pull request #32 from Social-projects-Rivne/feature/migrate_to_Sass_styling

Feature/migrate to sass styling

ltlmgw committed:
Added validation on user page, loader, fixed footer

ltlmgw committed:
footer fixes/404

ltlmgw committed:
changed fonts on user page

ltlmgw committed:
resolved conflicts

ltlmgw committed:
header footer fixes

ltlmgw committed:
fonts and links changes

Commit:
deleted superfluous image

iuriifaevskii committed:
resolve conflict with develop

Commit:
Merge pull request #34 from Social-projects-Rivne/profile/ui_fixes

Profile/ui fixes

iuriifaevskii committed:
fix conflict with develop

iuriifaevskii committed:
fix bug in login

iuriifaevskii committed:
Merge pull request #35 from Social-projects-Rivne/bugfix/login

fix bug in login

apasashkov committed:
removed user_id hardcore in users table for dbseed

l1s4un committed:
fig autoreload

DmytroNovik committed:
fixed style of search component

l1s4un committed:
Add orders and order_context tables to dbcreate.js

iuriifaevskii committed:
add functional switching type of search

l1s4un committed:
Merge pull request #37 from Social-projects-Rivne/feature/add_order_tables

Feature/add order tables

DmytroNovik committed:
added .idea to gitignore

DmytroNovik committed:
Update .gitignore

DmytroNovik committed:
deleted phpStorms files at all

DmytroNovik committed:
added config.example.js

DmytroNovik committed:
Merge pull request #39 from Social-projects-Rivne/feature/fix_search_drop_down

Feature/fix search drop down

l1s4un committed:
add buttons to each recipe card

ltlmgw committed:
loader changes

iuriifaevskii committed:
create page of list of all jobs

l1s4un committed:
add possibility add new order items to localStorage

iuriifaevskii committed:
moved dropdown out of form

iuriifaevskii committed:
Merge branch 'develop' into bugfix/search

iuriifaevskii committed:
Merge pull request #38 from Social-projects-Rivne/bugfix/search

bugfix/search

l1s4un committed:
change add logic

ltlmgw committed:
changed if empty name

iuriifaevskii committed:
Merge branch 'develop' of https://github.com/Social-projects-Rivne/Rv-026WebUI into feature/list_of_jobs

iuriifaevskii committed:
fix searchbar width

l1s4un committed:
add bucket icon to header

ltlmgw committed:
changed loader work. Name if field is empty.

iuriifaevskii committed:
Merge pull request #40 from Social-projects-Rivne/feature/list_of_jobs

Feature/list of jobs

ltlmgw committed:
res getUserInfo changes

ltlmgw committed:
styles changes

Commit:
Merge pull request #41 from Social-projects-Rivne/feature/profile_fixes

Feature/profile fixes

ltlmgw committed:
Merge branch 'develop' of https://github.com/Social-projects-Rivne/Rv-026WebUI into YanaAndriyYura

apasashkov committed:
added MultiSelect search option for ingredients

iuriifaevskii committed:
create the ability enter to the orderPage only for cooks and move orderList to common

iuriifaevskii committed:
Merge branch 'YanaAndriyYura' of https://github.com/Social-projects-Rivne/Rv-026WebUI into YanaAndriyYura

ltlmgw committed:
Merge branch 'YanaAndriyYura' of https://github.com/Social-projects-Rivne/Rv-026WebUI into YanaAndriyYura

iuriifaevskii committed:
add table orders_status and small fix showing all orders

l1s4un committed:
show cart content

iuriifaevskii committed:
add different display orderList for cook and user roles

apasashkov committed:
Added backend logic for search by ingredients

apasashkov committed:
removed handwritten.ttf and unneccessary multiselect.scss

apasashkov committed:
removed unecessary comments in css

StasMelnyk committed:
bugfix with editing recipe

fixed recipe controller
fixed common list
added test for editing recipe
changed test for viewing recipe

python4eg commented on config.js:
```diff
@@ -0,0 +1,17 @@
+    console.info('********');
+};
+
+export default {
```
Do You still use it?

iuriifaevskii committed:
create showing button to change the status depending on the role and status

StasMelnyk committed:
bugfix test

changed status from 500 to 404 if request is empty

apasashkov committed:
- change of server response in no ingredients are sent
- moved default and propTypes outside of the classes

ltlmgw committed:
'tabs'

ltlmgw committed:
user orders list

apasashkov committed:
fixed ternary operator assignment

iuriifaevskii committed:
create buttons to change the status and make order for defferent roles

ltlmgw committed:
tabs for users, changed components

iuriifaevskii committed:
create an opportynity to change the status color and color of the button depending on the status

iuriifaevskii committed:
added a test update

l1s4un committed:
add count to order's items

ltlmgw committed:
Merge branch 'YanaAndriyYura' of https://github.com/Social-projects-Rivne/Rv-026WebUI into YanaAndriyYura

ltlmgw committed:
created user page, changed qyery for orders by cooks and users

apasashkov committed:
Merge pull request #42 from Social-projects-Rivne/feature/search_by_ingredients

Feature/search by ingredients

l1s4un committed:
add order controllers and order model

l1s4un committed:
try to fix model errors

apasashkov committed:
- lazy load for getting all recipes
- minor changes in dbseed

apasashkov committed:
added search by ingredients' lazy load

apasashkov committed:
fixed footer and recipe card styles for lazy load

iuriifaevskii committed:
added the ability to edit statuses

iuriifaevskii committed:
added some styles and images to status massages

l1s4un committed:
fix updating cart in realtime

iuriifaevskii committed:
changed showing list of jobs with new fields in order and order_context

ltlmgw committed:
db changes

l1s4un committed:
change some styles

l1s4un committed:
change cart.js

ltlmgw committed:
status button changes

apasashkov committed:
change post to get request types when lazyloading

apasashkov committed:
Merge pull request #44 from Social-projects-Rivne/feature/lazyload

Feature/lazyload

iuriifaevskii committed:
added redux and test reducer

ltlmgw committed:
animation of avatar, table -> flex, button status

ltlmgw committed:
added slider, proptypes, changed query for orders

iuriifaevskii committed:
moved common files and add list of jobs link in header

iuriifaevskii committed:
resolved conflict and moved some common functional

iuriifaevskii committed:
fixed bug with key in orders

iuriifaevskii committed:
added flexbox to all orders

l1s4un committed:
add some animations, change cart  styles

l1s4un committed:
resolve conflicts

l1s4un committed:
fix error in dbcreate

l1s4un committed:
fix error after resolve conflicts

l1s4un committed:
fix one more eror

l1s4un committed:
Merge pull request #46 from Social-projects-Rivne/feature/create_order

Feature/create order

l1s4un committed:
fix style errors

iuriifaevskii committed:
fixed dbseed

l1s4un committed:
resolve conflicts

l1s4un committed:
Merge pull request #47 from Social-projects-Rivne/YanaAndriyYura

implement orders functionality

StasMelnyk committed:
Merge branch 'develop' into feature/editRecipe

StasMelnyk committed:
Merge pull request #43 from Social-projects-Rivne/feature/editRecipe

bugfix with editing recipe

l1s4un committed:
fix problem

iuriifaevskii committed:
add styles to order accept form

l1s4un committed:
Merge pull request #48 from Social-projects-Rivne/bugfix/cart

Bugfix/cart

iuriifaevskii committed:
Merge pull request #49 from Social-projects-Rivne/bugfix/order

add styles to order accept form

iuriifaevskii committed:
add animation to order status button

iuriifaevskii committed:
Merge pull request #50 from Social-projects-Rivne/bugfix/order_button_animation

add animation to order status button

iuriifaevskii committed:
pool a data from the develop

iuriifaevskii committed:
Merge pull request #45 from Social-projects-Rivne/feature/add_redux

To configure redux

iuriifaevskii committed:
add axios instead of fetch to recipes

iuriifaevskii committed:
add redux functional to recipes page

StasMelnyk committed:
fix problem with editing

fix problem with editing

StasMelnyk committed:
resolve conflict with edit

resolve conflict with edit

l1s4un committed:
add_init_cart

l1s4un committed:
connect to redux-devtools-extention

l1s4un committed:
change cartInit action (now it's work)

l1s4un committed:
show and hide cart with redux

l1s4un committed:
add items to cart with redux

iuriifaevskii committed:
Merge pull request #52 from Social-projects-Rivne/feature/recipes_with_redux

Feature/recipes with redux

l1s4un committed:
Merge pull request #51 from Social-projects-Rivne/feature/cart_with_redux

Feature/cart with redux

iuriifaevskii committed:
fix wrong http method in requests

iuriifaevskii committed:
rewrite the search by name and tag type on redux

apasashkov committed:
fix login without registration bug

ltlmgw committed:
added styles and some redux

Commit:
Merge pull request #53 from Social-projects-Rivne/style-fixes

added styles and some redux

iuriifaevskii committed:
pool dev changes

iuriifaevskii committed:
add constants to search

iuriifaevskii committed:
Merge pull request #54 from Social-projects-Rivne/bugfix/fix_and_clear_code

create search by name and tagtype via redux

iuriifaevskii committed:
fix ids in dbseed

iuriifaevskii committed:
Merge pull request #55 from Social-projects-Rivne/bugfix/dbseed_data

fix ids in dbseed

iuriifaevskii committed:
remove old webpack config

iuriifaevskii committed:
rename add order api url

iuriifaevskii committed:
add recipes are not found to search recipe page

iuriifaevskii committed:
fix bug. forbade the cook to take his order

iuriifaevskii committed:
change style custom file button

iuriifaevskii committed:
some fix in footer

iuriifaevskii committed:
pull new db create and db seed from Andriys branch

iuriifaevskii committed:
fix recipe description on search pages

iuriifaevskii committed:
Merge pull request #56 from Social-projects-Rivne/bugfix/global_fix

Bugfix/global fix

l1s4un committed:
fix cart styles, delete card's animation

iuriifaevskii committed:
add choice of role in registration

iuriifaevskii committed:
add role constants

iuriifaevskii committed:
Merge pull request #58 from Social-projects-Rivne/bugfix/choice_of_role

Bugfix/choice of role

l1s4un committed:
Merge pull request #57 from Social-projects-Rivne/bugfix/cart

Bugfix/cart

iuriifaevskii committed:
fix bug lazy load and clear some code

iuriifaevskii committed:
fix conflict

iuriifaevskii committed:
fix order for cook

iuriifaevskii committed:
add const status new

iuriifaevskii committed:
Merge pull request #59 from Social-projects-Rivne/bugfix/fix_lazyload

Bugfix/fix lazyload

ltlmgw committed:
styles fixes

l1s4un committed:
Add items to dbseed

iuriifaevskii committed:
add some data in dbseed

iuriifaevskii committed:
Merge pull request #60 from Social-projects-Rivne/bugfix/dbseed

Bugfix/dbseed
